Remotion/ObjectBinding/UnitTests/Validation/CompoundBusinessObjectPropertyPathBasedValidationFailureMatcherTest.cs

```diff
@@ -0,0 +1,171 @@
+      var firstResultObjectStub = new Mock<IBusinessObject>();
+      firstPropertyPathResultMock
+          .Setup(_ => _.ResultObject)
+          .Returns(firstResultObjectStub.Object);
+
+      var firstResultPropertyStub = new Mock<IBusinessObjectProperty>();
+      firstPropertyPathResultMock
+          .Setup(_ => _.ResultProperty)
+          .Returns(firstResultPropertyStub.Object);
+      var firstPropertyPathStub = new Mock<IBusinessObjectPropertyPath>();
+
+      firstPropertyPathStub
+          .Setup(_ => _.GetResult(rowObject.Object, BusinessObjectPropertyPath.UnreachableValueBehavior.ReturnNullForUnreachableValue, BusinessObjectPropertyPath.ListValueBehavior.FailForListProperties))
+          .Returns(firstPropertyPathResultMock.Object)
+          .Verifiable();
+
+      //second property
+      var secondPropertyPathResultMock = new Mock<IBusinessObjectPropertyPathResult>();
+      var secondResultObjectStub = new Mock<IBusinessObject>();
+      secondPropertyPathResultMock
+          .Setup(_ => _.ResultObject)
+          .Returns(secondResultObjectStub.Object);
+
+      var secondResultPropertyStub = new Mock<IBusinessObjectProperty>();
+      secondPropertyPathResultMock
+          .Setup(_ => _.ResultProperty)
+          .Returns(secondResultPropertyStub.Object);
+
+      var secondPropertyPathStub = new Mock<IBusinessObjectPropertyPath>();
+      secondPropertyPathStub
+          .Setup(_ => _.GetResult(rowObject.Object, BusinessObjectPropertyPath.UnreachableValueBehavior.ReturnNullForUnreachableValue, BusinessObjectPropertyPath.ListValueBehavior.FailForListProperties))
+          .Returns(secondPropertyPathResultMock.Object)
+          .Verifiable();
+
+      var validationErrors = new[]
+                             {
+                               BusinessObjectValidationFailure.Create("Oh no, something happened"),
+                               BusinessObjectValidationFailure.Create("Another error?"),
+                               BusinessObjectValidationFailure.Create("What are you doing?"),
+                             };
+
+      validationResultMock
+          .Setup(_ => _.GetValidationFailures(firstResultObjectStub.Object, firstResultPropertyStub.Object, true))
+          .Returns(new[] {validationErrors[0] })
+          .Verifiable();
+      validationResultMock
+          .Setup(_ => _.GetValidationFailures(secondResultObjectStub.Object, secondResultPropertyStub.Object, true))
+          .Returns(new[] { validationErrors[1], validationErrors[2] })
+          .Verifiable();
+
+      var matcher = new CompoundBusinessObjectPropertyPathBasedValidationFailureMatcher(new [] {firstPropertyPathStub.Object, secondPropertyPathStub.Object});
+      var result = matcher.GetMatchingValidationFailures(rowObject.Object, validationResultMock.Object);
+
+      Assert.That(result, Is.EqualTo(validationErrors));
+      firstPropertyPathStub.Verify();
+      secondPropertyPathStub.Verify();
+      validationResultMock.Verify();
```

Comment:
The same simplification recommendation applies in this test,too. We can make all of those into stubs.

Also, a stub should not (read: incorrect semantic naming) not have verifications. It could be a nice roslyn rule to that complains when you use Verification-methods with stub-named variables.